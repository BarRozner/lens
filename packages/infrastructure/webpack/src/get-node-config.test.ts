import { getNodeConfig } from "./get-node-config";

describe("get-node-config", () => {
  it("given an environment, works", () => {
    const actual = getNodeConfig({
      entrypointFilePath: "some-entrypoint-file-path",
      outputDirectory: "some-output-directory",
      environment: { mode: "development", devtool: "some-devtool" },
    });

    expect(actual).toMatchSnapshot();
  });
});
