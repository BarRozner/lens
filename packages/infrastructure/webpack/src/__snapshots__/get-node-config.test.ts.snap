// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-node-config given an environment, works 1`] = `
{
  "devtool": "some-devtool",
  "entry": {
    "index": "some-entrypoint-file-path",
  },
  "externalsPresets": {
    "node": true,
  },
  "mode": "development",
  "module": {
    "rules": [
      {
        "loader": "ts-loader",
        "test": /\\\\\\.ts\\(x\\)\\?\\$/,
      },
    ],
  },
  "name": "some-entrypoint-file-path",
  "node": {
    "__dirname": true,
    "__filename": true,
  },
  "output": {
    "filename": [Function],
    "library": {
      "type": "commonjs2",
    },
    "path": "some-output-directory",
  },
  "performance": {
    "hints": "error",
    "maxEntrypointSize": 100000,
  },
  "plugins": [
    MakePeerDependenciesExternalPlugin {},
    ProtectFromImportingNonDependencies {},
    ForkTsCheckerWebpackPlugin {
      "options": {
        "typescript": {
          "configOverwrite": {
            "compilerOptions": {
              "declaration": true,
              "declarationDir": "some-output-directory",
            },
            "include": [
              "some-entrypoint-file-path",
            ],
          },
          "mode": "write-dts",
        },
      },
    },
    LinkablePushPlugin {},
  ],
  "resolve": {
    "extensions": [
      ".ts",
      ".tsx",
      ".js",
    ],
  },
  "target": "node",
}
`;
